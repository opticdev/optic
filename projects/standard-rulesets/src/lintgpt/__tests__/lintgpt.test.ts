// import { expect, jest, test } from '@jest/globals';
// import { LintGpt, LintGptConfig } from '../index';
// import { OpenAPIV3 } from '@useoptic/openapi-utilities';
// import { TestHelpers } from '@useoptic/rulesets-base';
//
// jest.setTimeout(100000);
// test('lintgpt ', async () => {
//   const config: LintGptConfig = {
//     design: {
//       rules: [
//         'All operations must have an operationId',
//         'All operationIDs must be camelCase',
//         'operationId should NEVER change',
//         'The keys in an object property should all follow the same naming convention',
//         'GET operations should use 200 status code for its success response',
//         'All operations must have security defined',
//         'Arrays are not allowed to be the first thing in a response body. They MUST be wrapped like so {"users": []}',
//       ],
//     },
//   };
//   const prepared = await LintGpt.fromOpticConfig(config);
//
//   const beforeJson: OpenAPIV3.Document = {
//     ...TestHelpers.createEmptySpec(),
//     paths: {
//       '/api/users': {
//         get: {
//           operationId: 'getUsers',
//           responses: {
//             '201': {
//               description: 'Response from the API',
//               content: {
//                 'application/json': {
//                   schema: {
//                     type: 'array',
//                     items: {
//                       type: 'object',
//                       properties: {
//                         lastName: { type: 'string' },
//                         ID: { type: 'string' },
//                         first_name: { type: 'string' },
//                       },
//                     },
//                   },
//                 },
//               },
//             },
//           },
//         },
//       },
//     },
//   };
//   const afterJson: OpenAPIV3.Document = JSON.parse(JSON.stringify(beforeJson));
//   afterJson!.paths!['/api/users']!['get']!.operationId = 'get_all_the_users';
//
//   if (prepared instanceof LintGpt) {
//     const results = await TestHelpers.externalRulesWithInputs(
//       prepared,
//       beforeJson,
//       afterJson
//     );
//     expect(results).toMatchSnapshot();
//   }
// });
