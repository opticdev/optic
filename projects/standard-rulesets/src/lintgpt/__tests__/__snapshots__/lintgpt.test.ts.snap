// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lintgpt  1`] = `
[
  {
    "change": null,
    "error": undefined,
    "isMust": false,
    "isShould": true,
    "name": "All operations must have an operationId",
    "passed": true,
    "severity": 1,
    "where": "get /api/users",
  },
  {
    "change": null,
    "error": "The operationId 'get_all_the_users' is not in camelCase.",
    "isMust": false,
    "isShould": true,
    "name": "All operationIDs must be camelCase",
    "passed": false,
    "severity": 1,
    "where": "get /api/users",
  },
  {
    "change": null,
    "error": "The operationId has changed from 'getUsers' to 'get_all_the_users'. According to the rule, the operationId should never change.",
    "isMust": false,
    "isShould": true,
    "name": "operationId should NEVER change",
    "passed": false,
    "severity": 1,
    "where": "get /api/users",
  },
  {
    "change": null,
    "error": "The GET operation uses a 201 status code for its success response instead of the expected 200 status code.",
    "isMust": false,
    "isShould": true,
    "name": "GET operations should use 200 status code for its success response",
    "passed": false,
    "severity": 1,
    "where": "get /api/users",
  },
  {
    "change": null,
    "error": "The operation does not have security defined.",
    "isMust": false,
    "isShould": true,
    "name": "All operations must have security defined",
    "passed": false,
    "severity": 1,
    "where": "get /api/users",
  },
]
`;
